acme/examples/http01_example.py

```diff
@@ -0,0 +1,271 @@
+
+
+def perform_http01(client_acme, challb, orderr):
+    """Set up standalone webserver and perform HTTP-01 challenge."""
+
+    response, validation = challb.response_and_validation(client_acme.net.key)
+
+    resource = standalone.HTTP01RequestHandler.HTTP01Resource(
+        chall=challb.chall, response=response, validation=validation)
+
+    with challenge_server({resource}):
+
+        # This is the domain used for the local simple_verify. This example
+        # uses a fake domain name in DOMAIN, so this verification would fail
+        # because the DNS would never resolve it to a real IP address. In order
+        # to bypass it and only verify if the standalone server would
+        # successfully reply the correct resource, we use 'localhost' as domain
+        # instead.
+        domain = 'localhost'
+        if not response.simple_verify(challb.chall,
+                                      domain,
+                                      client_acme.net.key.public_key(),
+                                      PORT):
+            raise Exception('Verification failed')
+
+        # Let the CA server know that we are ready for the challenge.
+        client_acme.answer_challenge(challb, response)
+
+        # Wait for challenge status and then issue a certificate.
+        # It is possible to set a deadline time.
+        finalized_orderr = client_acme.poll_and_finalize(orderr)
+
+    return finalized_orderr.fullchain_pem
+
+
+# Main examples:
+
+
+def example_http():
+    """This example executes the whole process of fulfilling a HTTP-01
+    challenge for one specific domain.
+
+    The workflow consists of:
+    - Create account key.
+    - Register account and accept TOS.
+    - Select HTTP-01 within offered challenges by the CA server.
+    - Set up http challenge resource.
+    - Set up standalone web server.
+    - Create domain private key and CSR.
+    - Issue certificate.
+    - Change contact information.
+    - Renew Certificate
+    - Revoke Certificate
+    - Deactivate Account
+
+    """
+    logging.info('Example Challenge HTTP01')
+
+    client_acme = generate_client_account_key()
+
+    logging.info('Terms of Service URL: %s',
+                 client_acme.directory.meta.terms_of_service)
+
+    # Registration Resource: regr
+    # Creates account with contact information.
+    email = ('fake@example.com')
+    regr = client_acme.new_account(
+        messages.NewRegistration.from_data(
+            email=email, terms_of_service_agreed=True))
+
+    logging.info('Account registered.')
+
+    pkey_pem, csr_pem = new_csr_comp(DOMAIN)
+    orderr = client_acme.new_order(csr_pem)
+
+    challb = select_http01_chall(orderr)
+
+    # The certificate is ready to be used in the variable "fullchain_pem".
+    fullchain_pem = perform_http01(client_acme, challb, orderr)
+
+    logging.info('Certificate issued: \n%s', fullchain_pem)
+
+    regr = verify_registration(client_acme, regr)
+
+    # Change contact information
```

Comment:
Let's reorder so that account updates and certificate updates are grouped together

Reply:
No, I meant change the order to something like this:

(Account actions)
    - Create account key.
    - Register account and accept TOS.
    - Change contact information.
    - Deactivate Account
(Certificate actions)
    - Select HTTP-01 within offered challenges by the CA server.
    - Set up http challenge resource.
    - Set up standalone web server.
    - Create domain private key and CSR.
    - Issue certificate.
    - Renew Certificate
    - Revoke Certificate

But of course then if someone runs the code as is, they'll be deactivating an account they still need...

So maybe something like:
(Account creation)
    - Create account key.
    - Register account and accept TOS.
(Certificate actions)
    - Select HTTP-01 within offered challenges by the CA server.
    - Set up http challenge resource.
    - Set up standalone web server.
    - Create domain private key and CSR.
    - Issue certificate.
    - Renew Certificate
    - Revoke Certificate
(Account update actions)
    - Change contact information.
    - Deactivate Account

Reply:
Agreed, seems more organized now.,